runtime/interpreter/interpreter_expression.go

```diff
@@ -164,121 +167,188 @@ func (interpreter *Interpreter) evalExpression(expression ast.Expression) Value
 }
 
 func (interpreter *Interpreter) VisitBinaryExpression(expression *ast.BinaryExpression) ast.Repr {
+
+	leftValue := interpreter.evalExpression(expression.Left)
+
+	/* we make this a thunk so that we can skip computing it for certain short-circuiting
+	   operations */
+	rightValue := func() Value { return interpreter.evalExpression(expression.Right) }
```

Comment:
Good idea! Cuts down the repeated code 👍 

```suggestion
	// We make this a thunk so that we can skip computing it for certain short-circuiting operations
	rightValue := func() Value {
		return interpreter.evalExpression(expression.Right)
	}
```